Application/components/TextChannels/TextChannels.tsx

```diff
@@ -69,39 +69,46 @@ export function TextChannels() {
     }
   }, [user]); // Dependency array ensures this runs whenever `user` changes
 
-  const fetchUserChats = async () => {
-    if (!userID) return; // Ensure userID is available
-  
-    try {
-      const response = await fetch('/api/get-user-text-channels', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ userID: userID })
-      });
-  
-      if (response.ok) {
-        const { textChannels } = await response.json();
-        setTextChannels(textChannels);
-      } else {
-        console.error('Failed to fetch chat channels');
+  useEffect(() => {
+    const fetchUserChats = async () => {
```

Comment:
- In the fetchUserChats function, error handling is present for the fetch request, but there's no visual indication or user feedback when an error occurs. Consider implementing a notification system or error message display to inform users if there's a problem fetching chat channels.